src/analysis/inner_product_space/gram_schmidt_ortho.lean

```diff
@@ -90,116 +93,92 @@ begin
   right,
   cases hia.lt_or_lt with hia₁ hia₂,
   { rw inner_eq_zero_sym,
-    exact ih a h₀ i hia₁, },
-  { exact ih i hi a hia₂, },
+    exact ih a h₀ i hia₁ },
+  { exact ih i (mem_Ico.1 hi).2 a hia₂ }
 end
 
 /-- This is another version of `gram_schmidt_orthogonal` using `pairwise` instead. -/
-theorem gram_schmidt_pairwise_orthogonal (f : ℕ → E) :
+theorem gram_schmidt_pairwise_orthogonal (f : ι → E) :
   pairwise (λ a b, ⟪gram_schmidt 𝕜 f a, gram_schmidt 𝕜 f b⟫ = 0) :=
-@gram_schmidt_orthogonal 𝕜 _ _ _ f
+λ a b, gram_schmidt_orthogonal 𝕜 f
 
 open submodule set order
 
 /-- `gram_schmidt` preserves span of vectors. -/
-lemma span_gram_schmidt (f : ℕ → E) (c : ℕ) :
-  span 𝕜 (gram_schmidt 𝕜 f '' Iic c) = span 𝕜 (f '' Iic c) :=
+lemma span_gram_schmidt [succ_order ι] [is_succ_archimedean ι] (f : ι → E) (c : ι) :
+  span 𝕜 (gram_schmidt 𝕜 f '' Iio c) = span 𝕜 (f '' Iio c) :=
```

Comment:
Is `Iio c` definitely preferred to `Ico ⊥ c`? (Likewise `Iic c` vs `Icc ⊥ c` elsewhere.)

I have no strong feelings but I slightly prefer `Ico ⊥ c` because I immediately understand what I've got whereas the meaning of `Iio c` depends on whether the `c`'s type carries `order_bot`.

Reply:
This simp rule seems to suggest that `Iio c` is preferred: https://leanprover-community.github.io/mathlib_docs/data/set/intervals/basic.html#set.Ico_bot

Reply:
Aha, OK. Let's leave this then.

Reply:
> certain properties of `Iio c` depends on whether the `c`'s type carries `order_bot`.

Not once #14430 has landed. Even without it I agree with Alexander. `Ico ⊥ c` gets simped to `Iio c` (just as `⊥ ≤ x ∧ x < c` gets simped to `x < c`).